cardInfoAPI-Level1.yaml

```diff
@@ -0,0 +1,1402 @@
+openapi: 3.1.0
+info:
+  version: 0.7.0
+  title: Common Card API (Public)
+  description: >
+    This specification defines a common card API for payment cards used in Switzerland.
+    The API is supposed to be used by third parties to retrieve payment card information for scheme-based debit, credit or prepaid cards (read only).
+  contact:
+    email: info@common-api.ch
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+servers:
+  - url: https://card.common-api.ch
+externalDocs:
+  description: Find out more about SFTI API specifications
+  url: https://www.common-api.ch
+
+tags:
+  - name: Cards
+    description: Operations for retrieving information about payment cards.
+  - name: Card-tokens
+    description: Operations for retrieving information about tokens associated with payment cards.
+  - name: Transactions
+    description: Operations for retrieving information about transactions and authorizations done with payment cards or tokens.
+
+security:
+  - ApiKeyAuth: []
+  - OAuth2:
+      - read
+      - write
+
+paths:
+  /cards:
+    get:
+      summary: Provides a list of cards which match the provided query parameters.
+      description: >
+        This operation enables the retrieval of a list of payment cards. Payment cards can be physical or virtual cards.
+        In order to prevent API consumers from having to fulfill PCI DSS requirements, no full PAN or other data relevant for this regulation are included.
+        The API consumer is expected to have a non-card related identifier for the customer for whom data should be returned (e.g. card account number, bank account number).
+      tags:
+        - Cards
+      parameters:
+        - in: query
+          name: contract_reference
+          required: false
+          schema:
+            $ref: '#/components/schemas/ContractReference'
```

Comment:
Exposure in URLs: ContractReference, PersonReference, CardAccountReference, BankAccountReference are considered to be customer identifiers. Query parameters and path segments are often logged by web servers, proxies, and client browsers. This can lead to unintended exposure of sensitive information.

Use the HTTP Body for Sensitive Data instead. `POST /cards/search` to sumbit the search criteria as a custom action (imperative verb: search)

Reply:
Thanks for your feedback. We will remove the query parameters which refer to customer identifiers (*_reference attributes). At least from the level 1. For level 2 we will probably add query parameters again, but with the ids, not the references.
The GET operation will for level 1 return all cards which are allowed to be accessed with the consent that the API user is providing. The only remaining filter attribute will be card_status. Other filtering has to be done at the API user.
One impact from this change is that the consent flow has to make sure that the third party knows for which customers he can access data (matching between user of third party and card). It should still not be necessary that third parties handle PANs. An attribute for the consent will be added based on the ongoing definitions of the ca-security.